src/back/TicketManager/tests/PaymentService/TicketManager.PaymentService.ServicesTests/Endpoints/Payments/CheckPaymentStatusEndpointTest.cs

```diff
@@ -0,0 +1,38 @@
+using FastEndpoints;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using Moq.EntityFrameworkCore;
+using TicketManager.PaymentService.Contracts.Payments;
+using TicketManager.PaymentService.Domain.Payments;
+using TicketManager.PaymentService.Services.Configuration;
+using TicketManager.PaymentService.Services.DataAccess;
+using TicketManager.PaymentService.Services.Endpoints.Payments;
+using Xunit;
+
+namespace TicketManager.PaymentService.ServicesTests.Endpoints.Payments;
+
+public class CheckPaymentStatusEndpointTest
+{
+    [Fact]
+    public async Task WhenItIsCalled_ItShouldAddPayment()
+    {
+        List<Payment> payments = new List<Payment> {new(), new(), new()};
+        
+        
+        var dbContextMock = new Mock<PaymentDbContext>(new DbContextOptionsBuilder<PaymentDbContext>().Options);
+        dbContextMock.Setup(d => d.Payments).ReturnsDbSet(payments);
+        var dbContext = dbContextMock.Object;
+        var config = new PaymentServiceConfiguration("ApiKey");
+        var endpoint = Factory.Create<CheckPaymentStatusEndpoint>(dbContext, config);
+
+        var secondPayment = payments[1];
+        var request = new CheckPaymentStatusRequest { Id = secondPayment.Id };
+        var expected = secondPayment.PaymentStatus;
+        
+        await endpoint.HandleAsync(request, default);
+        var result = endpoint.Response;
+        
+        result.Status.Should().Be((PaymentStatusDto)expected);
```

Comment:
If you use `BeEquivalentTo`, you don't need to convert.

Tip for the future: use `BeEquivalentTo` to every type that does not have overloaded `==` operator and types that are not the same type, but have the same fields.

For example you have
```
var a = new A { a = 1, b = 2, };
var b = new B { b = 2, c = 3, };
```
Then `a.Should().BeEquivalentTo(b)` passes.

Reply:
Especially `BeEquivalentTo` is very good for lists assertions.

Reply:
From what I see Should() for enums does not have this method but thanks for the tip.